scopes/component/component-descriptor/aspect-map.ts

```diff
@@ -0,0 +1,33 @@
+export type AspectDataEntry = {
+  aspectId: string;
+  aspectData: string;
+};
+
+export type AspectMapProps = {
+  entries?: AspectDataEntry[];
+};
+
+export class AspectMap {
+  constructor(private entries: AspectDataEntry[]) {}
+
+  get<T>(aspectId: string): T | undefined {
+    if (!this.entries || !Array.isArray(this.entries)) return undefined;
+    const aspectEntry = this.entries.find((entry) => entry.aspectId === aspectId);
+    if (!aspectEntry) return undefined;
+    if (typeof aspectEntry.aspectData === 'string') {
+      return JSON.parse(aspectEntry.aspectData) as T;
```

Comment:
@guysaar223 we need to check this is ok. aspect data is not always a string so it breaks when running `JSON.parse` on it

Reply:
added the `if` for now to prevent this issue